game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -85,16 +85,19 @@ public void load(final URI uri) {
     fileName = null;
     this.gameData = parseAndValidate(uri);
     if (gameData == null || gameData.getGameName() == null || uri == null) {
-      ClientSetting.defaultGameName.resetValue();
-      ClientSetting.defaultGameUri.resetValue();
-      this.fileName = "-";
-      this.gameName = "-";
-      this.gameRound = "-";
+      this.resetDefaultGame();
     } else {
       setGameData(gameData);
       ClientSetting.defaultGameName.setValue(gameData.getGameName());
       ClientSetting.defaultGameUri.setValue(uri.toString());
+      ClientSetting.flush();
     }
+  }
+
+  private void resetDefaultGame() {
+    setGameData(null);
```

Comment:
This set game data seems to be an unnecessary store coming from line 88. It's also not consistent as the other lines sets the game data directly and this uses a setter method. Generally, AFAIK it's better to avoid setter methods within a class (lest someone overrides your setter method and then suddenly they've changed a lot more code then expected).

2nd, the abstraction of the method above seems to be kinda mixed. It's basically:
```
low level checks
low level setting
low level checks

if( low level condition)
    do medium level
else
   do low level
```

I recommend to accept a `GameData` parameter to this method and then you can re-use. Arguably it's a method so that the code can be re-used, but we still have the same calls on lines 91 and 92 (it's not the same, but the data we set is the same, that would be the idea behind DRY, the information of which values to set is repeated in 2 places).

As a method, it would be:
```
private void setDefaultGame(@Nullable final URI fileUri, @Nullable final GAmeData gamedata) {
  if(gameData == null) {
    ClientSetting.defaultGameName.resetValue();
    ClientSetting.defaultGameUri.resetValue();  
  } else {
      ClientSetting.defaultGameName.setValue(gameData.getGameName());
      ClientSetting.defaultGameUri.setValue(uri.toString());
  }
   ClientSetting.flush();
}

```

Now, looking at the above, the fact we have a decoupled file uri and game name is really not good too. That looks very much like a data clump code smell and would do well to be unified behind a single object or a single data type. I think that is probably best resolved as a follow up where we stop storing the default game name, just the default uri, and lookup the game name from that URI when it is needed.

Reply:
I made this change.